video/main.py

```
@@ -1,24 +1,127 @@
+
+        Returns the total number of faces detected across all sampled frames.
+        """
+        cap = cv2.VideoCapture(path)
+        frame_idx = 0
+        faces_total = 0
+        while True:
+            ret, frame = cap.read()
+            if not ret:
+                break
+            if frame_idx % 30 == 0:  # Sample every 30th frame
+                temp = tempfile.NamedTemporaryFile(suffix=".jpg", delete=False)
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)